src/core/bullet.js

```diff
@@ -0,0 +1,57 @@
+import { Sprite, Assets } from "pixi.js";
+
+export class bulletProjectile {
+  constructor(bulletX, bulletY, app) {
+    this.bulletX = bulletX;
+    this.bulletY = bulletY;
+    this.app = app;
+
+    // when sliders are implemented, these values should be manipulated roughly 
+    this.bulletSpeedX = 100;
+    this.bulletSpeedY = 50;
+    this.gravity = 5;
+  }
+
+  getX() {
+    return this.bulletX;
+  }
+
+  getY() {
+    return this.bulletY;
+  }
+
+  async initialiseSprite() {
+    // load texture of player, and convert into sprite.
+    const texture = await Assets.load('assets/images/bullet.png'); // 'await' keyword used for asynchronous texture loading
+    const sprite = Sprite.from(texture);
+    sprite.anchor.set(0.5, 0.5); // set anchor point to centre of the sprite
+
+    // initialise x, y to arguements passed through via constructor
+    sprite.x = this.bulletX;
+    sprite.y = this.bulletY;
+    sprite.width = 30;
+    sprite.height = 30;
+    this.sprite = sprite;
+  }
+
+  getSprite() {
+    if (this.sprite) {
+      console.log("Sprite initialised!");
+      return this.sprite;
+    } else {
+      console.log("Sprite not initialised!");
```

Comment:
Should just error if the user can't shoot no?
If you don't want it to error then we should add better handling eg the option to shoot again

Reply:
You are right, but these console log commands were mostly just used for testing and debugging. I can get rid of them if needed to clean the code up